kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -281,6 +293,38 @@ private SslContextFactory getSSLContextFactory() {
         return  sslContextFactory;
     }
 
+    /**
+     * Creates a Handler instance to handle incoming HTTP requests for readiness check
+     *
+     * @return Handler
+     */
+    /* test */ Handler getReadinessHandler() {
+        return new AbstractHandler() {
+            @Override
+            public void handle(String s, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
+                response.setContentType("application/json");
+                response.setCharacterEncoding("UTF-8");
+                baseRequest.setHandled(true);
+                if (brokerState != null) {
+                    byte observedState = (byte) brokerState.value();
+                    boolean stateIsRunning = BROKER_RUNNING_STATE <= observedState && BROKER_UNKNOWN_STATE != observedState;
+                    if (stateIsRunning) {
+                        LOGGER.trace("Broker is in running according to {}. The current state is {}", brokerStateName, observedState);
+                        response.setStatus(HttpServletResponse.SC_OK);
```

Comment:
We are not bringing a body in the HTTP response, I would use `204 NO CONTENT` instead of OK (i.e. this is what we have in our Strimzi HTTP bridge on the readiness/liveness probes).